### ISSUE TITLE
UI: Required filed of type "array" is ignored

### ISSUE THREAD

**Issue by artem1205** [airbytehq/airbyte#24092]
## Environment
- **Airbyte version**:  0.42.0
- **OS Version / Instance**:macOS
- **Deployment**: docker
- **Source Connector and version**: any
- **Step where error happened**: Setup new connection / Update connector

## Current Behavior
Required filed of type "array" is ignored.
Using the provided configuration, I was able to pass UI checks, and error raised only after click on test button, with not very useful error:
```
  "connectionSpecification": {
    "$schema": "https://json-schema.org/draft-07/schema#",
    "title": "OneSignal Source Spec",
    "type": "object",
    "required": ["user_auth_key", "start_date", "outcome_names", "applications"],
    "additionalProperties": true,
    "properties": {
      "user_auth_key": {
        "type": "string",
        "title": "User Auth Key",
        "description": "OneSignal User Auth Key, see the <a href=\"https://documentation.onesignal.com/docs/accounts-and-keys#user-auth-key\">docs</a> for more information on how to obtain this key.",
        "airbyte_secret": true,
        "order": 0
      },
      "applications": {
        "type": "array",
        "title": "Applications",
        "minItems": 1,
```
![image](https://user-images.githubusercontent.com/36314070/225352350-a759f9fd-e124-49d2-acf8-3f0f83f7deed.png)
![image](https://user-images.githubusercontent.com/36314070/225352760-3c897a86-85ae-4955-a4a3-52fb8ad9f5df.png)

## Expected Behavior
"minItems": parameters should be handled as others limitations

## Steps to Reproduce
1. Set  `"minItems": 1` to any field in config with type: array
2. Create new source with empty array

**Comment by timroes** [comment]
cc @flash1293 @lmossman

**Comment by flash1293** [comment]
Yeah, we don't use the "minItems" property in the client side validation (see also https://github.com/airbytehq/airbyte/issues/20064 ), so it falls back to the connector-side validation that captures it correctly. If this is blocking some important feature I can take care of it, it seems like a small change.

**Comment by lmossman** [comment]
`minItems` should be supported for all array fields:
- Add to the schemaToYup helper
- Errors should be handled and shown for each of the following component types in the airbyte UI:
  - Arrays of objects (ArraySection)
  - Enum arrays (Multiselect)
  - Non-enum primitive-type arrays (TagInput)
